pkg/cfn/manager/nodegroup.go

```diff
@@ -215,6 +215,13 @@ func (c *StackCollection) GetUnmanagedNodeGroupAutoScalingGroupName(ctx context.
 	if err != nil {
 		return "", err
 	}
+	switch res.StackResourceDetail.ResourceStatus {
+	case types.ResourceStatusCreateFailed, types.ResourceStatusDeleteComplete, types.ResourceStatusDeleteInProgress:
+		return "", fmt.Errorf("\"%s\" resource of stack \"%s\" is in \"%s\" state", *input.LogicalResourceId, *s.StackName, res.StackResourceDetail.ResourceStatus)
```

Comment:
You can use the `%q` formatting verb.

```suggestion
		return "", fmt.Errorf("%q resource of stack %q is in %q state", *input.LogicalResourceId, *s.StackName, res.StackResourceDetail.ResourceStatus)
```